test/tests/functional/pbs_advance_resv_end_hook.py

```diff
@@ -0,0 +1,169 @@
+    """
+
+    def setUp(self):
+        """ Create and import the python script for reservation end hook
+            in setup method"""
+
+        super(TestAdvanceReservationEndHook, self).setUp()
+        hook_script = """
+import pbs
+e=pbs.event()
+
+pbs.logmsg(pbs.LOG_DEBUG, 'Reservation End Hook name - %s' % (pbs.event().hook_name))
+
+if e.type == pbs.RESV_END:
+    pbs.logmsg(pbs.LOG_DEBUG, 'Reservation ID - %s'%pbs.event().resv.resvid)
+
+e.reject()
+"""
+        self.hook_name = 'resvend_hook'
+        attrs = {'event': 'resv_end'}
+        self.server.create_import_hook(
+            self.hook_name, attrs, hook_script, overwrite=True)
+
+        self.server.manager(
+            MGR_CMD_SET, SERVER, {
+                'log_events': 2047}, expect=True)
+
+    def submit_and_confirm_resv(self, offset, duration, select="1:ncpus=1"):
+        """ Submit and confirm the advance reservation """
+
+        start_time = time.time() + offset
+        end_time = start_time + duration
+        attrs = {
+            'reserve_start': start_time,
+            'reserve_end': end_time,
+            'Resource_List.select': select}
+        rid = self.server.submit(Reservation(TEST_USER, attrs))
+
+        resv_attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, resv_attrs, id=rid)
+
+        return rid
+
+    def test_delete_resv(self):
+        """ Test to delete a confirmed reservation
+            and verify the resvend hook. """
+
+        offset = 3
+        duration = 20
+        rid = self.submit_and_confirm_resv(offset, duration)
+
+        self.server.delete(rid)
+        msg = "Hook;Server@" + self.server.hostname.lower() + ";Reservation ID - %s" % rid
+        self.server.log_match(msg, tail=True, interval=2, max_attempts=30)
+
+    def test_delete_degraded_resv(self):
+        """ Test to delete a degraded reservation
+            and verify the resvend hook. Single node setup testcase"""
+
+        offset = 3
+        duration = 60
+        rid = self.submit_and_confirm_resv(offset, duration)
+
+        self.mom.stop()
+
+        msg = 'mom is not down'
+        self.assertFalse(self.mom.isUp(), msg)
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10')}
+        self.server.expect(RESV, attrs, id=rid)
+
+        self.server.delete(rid)
+        msg = "Hook;Server@" + self.server.hostname.lower() + ";Reservation ID - %s" % rid
+        self.server.log_match(msg, tail=True, interval=2, max_attempts=30)
+
+    def test_delete_resv_server_down(self):
+        """ Test to delete a confirmed reservation and make the server off and on
+            and verify the resvend hook. """
+
+        offset = 3
+        duration = 60
+        rid = self.submit_and_confirm_resv(offset, duration)
+
+        self.server.stop()
+
+        msg = 'server is not down'
+        self.assertFalse(self.server.isUp(), msg)
+
+        self.server.start()
+        self.logger.info(msg)
```

Comment:
I'm not sure this log message is right.  You're about to assert that the server is up.  You will log this regardless of that assert.